src/Data/Text.hs

```diff
@@ -362,6 +362,16 @@ instance Read Text where
 instance Semigroup Text where
     (<>) = append
 
+    -- | Beware: this function will evaluate to error if the given number does
+    -- not fit into an @Int@.
```

Comment:
Sadly it turns out Haddock does not see comments to instance methods. I looked at the documentation generated by `cabal haddock` — this comment is not rendered. This also seems to be confirmed on the Internet.

* <https://stackoverflow.com/questions/17758681/haddock-documentation-for-instance-functions-with-quirks-replaced-by-default-cl>
* https://github.com/haskell/haddock/issues/123

I am going to move this comment to the instance.